src/vulkan/compute_pipeline.cc

```diff
@@ -68,12 +68,31 @@ Result ComputePipeline::Compute(uint32_t x, uint32_t y, uint32_t z) {
   if (!r.IsSuccess())
     return r;
 
+  r = command_->End();
+  if (!r.IsSuccess())
+    return r;
+
+  r = command_->SubmitAndReset(GetFenceTimeout());
+  if (!r.IsSuccess())
+    return r;
+
   if (pipeline_ == VK_NULL_HANDLE) {
     Result r = CreateVkComputePipeline();
     if (!r.IsSuccess())
       return r;
   }
 
+  // Note that a command updating a descriptor set and a command using
+  // it must be submitted separately, because using a descriptor set
+  // while updating it is not safe.
+  r = UpdateDescriptorSetsIfNeeded();
+  if (!r.IsSuccess())
+    return r;
+
+  r = command_->BeginIfNotInRecording();
+  if (!r.IsSuccess())
+    return r;
+
   r = UpdateDescriptorSetsIfNeeded();
```

Comment:
Is this still needed? It's weird we call UpdateDescriptorSetsIfNeeded before and after the BeginfNotInRecording.

Reply:
Nice catch. This is my mistake. Removed